### PR TITLE
Implemented PXB-2407 - Merge MySQL 5.7.34

### PR DESCRIPTION
**Pull request by altmannmarcelo**
https://jira.percona.com/browse/PXB-2407

Merge of upstream 5.7.34

### CODE REVIEW

**Comment by altmannmarcelo** [comment]
https://pxb.cd.percona.com/job/percona-xtrabackup-2.4-test-param/31/

File: storage/innobase/log/log0recv.cc [hunk 1]
```diff
@@ -3962,11 +3962,7 @@ recv_group_scan_log_recs(
 	bool		last_phase)
 {
 	DBUG_ENTER("recv_group_scan_log_recs");
-<<<<<<< HEAD
-//	DBUG_ASSERT(!last_phase || recv_sys->mlog_checkpoint_lsn > 0);
-=======
-	assert(!last_phase || recv_sys->mlog_checkpoint_lsn > 0);
->>>>>>> mysql-5.7.34
+//	assert(!last_phase || recv_sys->mlog_checkpoint_lsn > 0);
```
Comment (satya-bodapati) [inline storage/innobase/log/log0recv.cc]: can this be uncommented? (HEAD enabled this assert)

File: mysql-test/CMakeLists.txt [hunk 2]
```diff
@@ -55,109 +57,108 @@ INSTALL(
...
+      ${CMAKE_CURRENT_BINARY_DIR}/mysql-test-run.pl
+      @ONLY
+    )
+  ENDIF()
+  IF(UNIX)
+    EXECUTE_PROCESS(
+      COMMAND chmod +x  mysql-test-run.pl
+      COMMAND ${CMAKE_COMMAND} -E create_symlink 
+       ./mysql-test-run.pl mtr
+      COMMAND ${CMAKE_COMMAND} -E create_symlink
```
Comment (satya-bodapati) [inline mysql-test/CMakeLists.txt:75]: this entire change in file is only because of the new IF block?

**Review by satya-bodapati — CHANGES_REQUESTED** [review]

File: mysql-test/CMakeLists.txt (hunk 2 above)
Comment (altmannmarcelo) [reply mysql-test/CMakeLists.txt:75]: @satya-bodapati yes, I've adjusted formatting, so the extra tab caused all this. But the only change is https://github.com/percona/percona-xtrabackup/pull/1093/files/33539bb93f966798637676150098f7569f529efa..368a7de038823cc15b89e4736164a4499d0358bf#diff-25b9cb7fd9753fffb13a4219e2f67eb22373660c6aaf7126eaa0a8442001ec6eR25 and https://github.com/percona/percona-xtrabackup/pull/1093/files/33539bb93f966798637676150098f7569f529efa..368a7de038823cc15b89e4736164a4499d0358bf#diff-25b9cb7fd9753fffb13a4219e2f67eb22373660c6aaf7126eaa0a8442001ec6eR164

File: storage/innobase/log/log0recv.cc (hunk 1 above)
Comment (altmannmarcelo) [reply storage/innobase/log/log0recv.cc]: This is currently commented out in 2.4 code - https://github.com/percona/percona-xtrabackup/blob/2.4/storage/innobase/log/log0recv.cc#L3965

**Review by satya-bodapati — APPROVED** [review]
All the 48 test failures are mdl_locks.sh  We should either fix the test or remove it. what do u say @altmannmarcelo

**Comment by altmannmarcelo** [comment]
@satya-bodapati thanks for reviewing. Regarding mdl_lock I will check what is needed to fix it. Same noise is happening on 8.0

**Pull request merged**